packages/@aws-cdk/aws-servicecatalog/lib/portfolio.ts

```diff
@@ -0,0 +1,370 @@
+      tagOption[key].forEach(value => {
+        InputValidator.validateLength('value', value.value, 1, 256);
+      });
+    });
+    AssociationManager.associateTagOption(this, this.portfolioId, tagOption);
+  }
+
+  /**
+   * Create a unique id based off the L1 portfolio.
+   *
+   */
+  protected abstract getKeyForPortfolio(value: string): string;
+}
+
+/**
+ * Properties for a Portfolio.
+ */
+export interface PortfolioProps {
+
+  /**
+     * Enforces a particular physical portfolio name.
+     * @default <generated>
+     */
+  readonly portfolioName: string;
+
+  /**
+     * The provider name.
+     *
+     */
+  readonly providerName: string;
+
+  /**
+     * The accept language.
+     * @default
+     */
+  readonly acceptLanguage?: AcceptLanguage;
+
+  /**
+     * Description for portfolio.
+     *
+     * @default
+     */
+  readonly description?: string;
+
+
+  /**
+     * A collection of tags attached to portfolio.
+     * @default
+     *
+     */
+  readonly tags?: Tag[];
+
+}
+
+/**
+ * A Service Catalog portfolio.
+ */
+export class Portfolio extends PortfolioBase {
+
+  /**
+     * Creates a Portfolio construct that represents an external stream.
+     *
+     * @param scope The parent creating construct (usually `this`).
+     * @param id The construct's name.
+     * @param attrs Portfolio import properties
+     */
+  public static fromPortfolioAttributes(scope: Construct, id: string, attrs: PortfolioAttributes): IPortfolio {
+    class Import extends PortfolioBase {
+      public readonly id = id
+      public readonly portfolioId = attrs.portfolioArn.split('/').pop()!
+      public readonly portfolioArn = attrs.portfolioArn;
+      public readonly portfolioName = attrs.portfolioName;
+
+      protected getKeyForPortfolio(value: string): string {
+        return getIdentifier(this.portfolioArn, value);
+      }
+    }
+
+    return new Import(scope, id);
+  }
+
+  public readonly portfolioArn: string;
+  public readonly portfolioName: string;
+  public readonly portfolioId: string;
+
+  private readonly portfolio: CfnPortfolio;
+
+  constructor(scope: Construct, id: string, props: PortfolioProps) {
+    super(scope, id, {
+      physicalName: props.portfolioName,
+    });
+
+    this.validatePortfolioProps(props);
+
+    this.portfolio = new CfnPortfolio(this, 'Resource', {
+      displayName: props.portfolioName,
+      providerName: props.providerName,
+      description: props.description,
+      tags: props.tags,
+      acceptLanguage: props.acceptLanguage,
+    });
+
+    this.portfolioArn = this.getResourceArnAttribute(this.portfolio.ref, {
+      service: 'catalog',
+      resource: 'portfolio',
+      resourceName: this.portfolio.ref,
+    });
+
+    this.portfolioName = this.getResourceNameAttribute(props.portfolioName);
+
+    this.portfolioId = this.getResourceNameAttribute(this.portfolio.ref);
+
+  }
+
+  protected getKeyForPortfolio(value: string): string {
```

Comment:
we need to override this for our 2 different cases,

`Import` will not have the L1 for the node address for the construct ID,

and `Portfolio` won't have the arn at compile time as it will be a Token